electron/IPC/codeGenerator.ts

```diff
@@ -53,183 +62,200 @@ export default codeGenearator;
+            await fileSystem.writeFile(metaFilePath, JSON.stringify(metaData, null, 2));
+        }
+    } catch (err) {
+        actionRs = { isSuccess: false, errorMessage: String(err) };
     }
 
-    // Actions are completed, let's Save error (if any) to the .code-metadata
-    if (!actionRs.isSuccess) {
-      const metaFilePath = path.join(rmprojFile.folderPath, StandardFolder.OutputCode, StandardOutputFile.MetaData);
-      const metaData = JSON.parse(await fileSystem.readFile(metaFilePath)) as ITestSuitesMetadata;
-      metaData.error = { message: actionRs.errorMessage ?? "Unknown error" };
-      await fileSystem.writeFile(metaFilePath, JSON.stringify(metaData, null, 2));
+    const ipcRs: IIpcGenericResponse<{ logFile: string }> = {
+        isSuccess: actionRs.isSuccess,
+        errorMessage: actionRs.errorMessage,
+    };
+
+    // Print log file
+    try {
+        sb.appendLine(`*** Finish at ${moment().format("MMMM Do YYYY, h:mm:ss a")}`);
+        sb.appendLine(JSON.stringify(ipcRs, null, 4));
+
+        const logsFolder = path.join(rmprojFile.folderPath, StandardFolder.Logs);
+        await fileSystem.createFoler(logsFolder);
+
+        const logFileName = `gen-code.${moment(startTime).format("YYYYMMDD_HHmmss")}.log`;
+        const logFilePath = path.join(logsFolder, logFileName);
+        await fileSystem.writeFile(logFilePath, sb.toString());
+
+        ipcRs.data = { logFile: path.join(StandardFolder.Logs, logFileName) };
+    } finally {
+        browserWindow.webContents.send("finish", ipcRs);
+    }
+}
+
+async function generateProjectMetadata(
+    browserWindow: BrowserWindow,
+    event: Electron.IpcMainEvent,
+    projectFile: IRmProjFile
+): Promise<IIpcGenericResponse<ITestSuitesMetadata>> {
+    try {
+        const actionRs = await genProjectMetadata(projectFile);
+        return { isSuccess: true, data: actionRs.data } as IIpcResponse;
+    } catch (error) {
+        return { isSuccess: false, errorMessage: "Cannot generate suites metadata: " + error };
     }
-  } catch (err) {
-    actionRs = { isSuccess: false, errorMessage: String(err) };
-  }
-
-  const ipcRs: IIpcGenericResponse<{ logFile: string }> = {
-    isSuccess: actionRs.isSuccess,
-    errorMessage: actionRs.errorMessage,
-  };
-
-  // Print log file
-  try {
-    sb.appendLine(`*** Finish at ${moment().format("MMMM Do YYYY, h:mm:ss a")}`);
-    sb.appendLine(JSON.stringify(ipcRs, null, 4));
-
-    const logsFolder = path.join(rmprojFile.folderPath, StandardFolder.Logs);
-    await fileSystem.createFoler(logsFolder);
-
-    const logFileName = `gen-code.${moment(startTime).format("YYYYMMDD_HHmmss")}.log`;
-    const logFilePath = path.join(logsFolder, logFileName);
-    await fileSystem.writeFile(logFilePath, sb.toString());
-
-    ipcRs.data = { logFile: path.join(StandardFolder.Logs, logFileName) };
-  } finally {
-    browserWindow.webContents.send("finish", ipcRs);
-  }
 }
 
 async function getSuitesMetadata(
```

Comment:
Rename this method (and other related stuff), from `getSuitesMetadata` to `getProjectMetadata`